File: Data Structures & Algorithms/maximum-subarray/submission-2.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        int curSum = 0;

        for (int n : nums){
            curSum = Math.max(0, curSum);
            curSum += n;
            maxSum = Math.max(curSum, maxSum);
        }
        return maxSum;
    }
}

File: Data Structures & Algorithms/max-consecutive-ones/submission-2.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max_ = 0
        i = 0
        if len(nums) == 1:
            return nums[0] 
        count = 0
        while i < len(nums):
            if nums[i] == 1:
                count += 1
            else:
                count = 0
            max_ = max(max_, count)
            i+=1
        return max_